### DISCUSSION TITLE
Problem with home row mods and layer switching

### DISCUSSION THREAD

**Discussion by oysandvik94** [getreuer/qmk-keymap#56] [General]
Hi, thanks for a great project!

I have a weird issue where I have a code LT(1, KC_DEL). Layer 1 has a keycode KC_EXCL. On layer 0, this same key is a LALT_T(KC_R). 

When I press the LT key in order to press KC_EXCL, I get the shifted version, the number 1. 

This is only the first time. If I hold the LT button a bit before I press !, it works as expected.

So there seems to be some timeout stuff that is happening. If I replace LALT_T(KC_R) with KC_R, I dont have any issuse.

I am not sure why there it is getting shifted, since removing the alt modifier fixes the issue.

The code I am using is here:

https://github.com/oysandvik94/qmk_firmware/tree/master/keyboards/beekeeb/piantor_pro

Anyone have any tips to what I can try? I've experimented with a lot of options, but havent seen any differences. I'll try to debug the achordion code later maybe, but I am not a C person, so it will be an uphill battle.

**Comment by akaralar** [comment]
@oysandvik94 This is a shot in the dark but do you happen to be using Karabiner by any chance? I had a similar issue with a key getting shifted while it shouldn't and turning off Karabiner fixed it for me.

**Reply by oysandvik94** [reply]
No, not completely sure what Karabiner is. But if it's external software then I can maybe try a different machine to see it I get the same results.

**Reply by akaralar** [reply]
If I understand correctly, your `LT(1, KC_DEL)` is on the thumb right? Have you tried disabling Achordion for thumb keys as described here? https://getreuer.info/posts/keyboards/achordion/index.html#achordion_chord

**Reply by oysandvik94** [reply]
Yes, I messed a little bit around with that. 

That would make it so that I can use a thumb key + another key on the same half right? I managed to do that after some forth and back, but I'm pretty sure I still had the same issue. Also, I do want to keep achordian on thumb keys anyway.

**Comment by oysandvik94** [comment]
I actually found that issue happens without Achordion as well, but the delay needed to get the correct symbol is shorter. So it seems there is some timeout or setting that Achordion changes that exposes this bug I have. So probably not Achordions fault. 

Also, it only happens in alacritty?
